src/object/object_primitive.c

```diff
@@ -11132,35 +11132,9 @@ mr_cmpval_string (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, int tot
       return DB_UNK;
     }
 
-  if (!ignore_trailing_space)
+  if (ignore_trailing_space)
     {
-      int i;
-
-      if (type1 == DB_TYPE_CHAR || type1 == DB_TYPE_NCHAR)
-	{
-	  for (i = size1; i > 1; i--)
-	    {
-	      if (string1[i - 1] != 0x20)
-		{
-		  break;
-		}
-	    }
-	  size1 = i;
-	}
-
-      if (type2 == DB_TYPE_CHAR || type2 == DB_TYPE_NCHAR)
-	{
-	  for (i = size2; i > 1; i--)
-	    {
-	      if (string2[i - 1] != 0x20)
-		{
-		  break;
-		}
-	    }
-	  size2 = i;
-	}
-
-      ti = false;
+      ti = true;
```

Comment:
We can now remove ti and use ignore_trailing_space instead.

Reply:
ignore_trailing_space is a static global and the variable is set only once, however, the variable ti is set whenever this routine called. Therefore ti has a role to apply comparison method.

Reply:
Ok, I refactored the code as your comment.